Microsoft/Git-Credential-Manager-for-Mac-and-Linux
# Add support for GNOME Keyring

Description
-----------
The bulk of the work to support the GNOME Keyring on Linux was to adopt the `auth-secure-storage` module of the [Visual Studio Team Services Authentication Library for Java](https://github.com/Microsoft/vsts-authentication-library-for-java), removing a lot of the code that was effectively moved to the library's modules.

Manual testing
--------------

1. On a Fedora 22 machine, configure Git to use the output of this branch's build.
2. Find a Git remote hosted in a Visual Studio Team Services account and perform a `git fetch` from it.
    1. I already had a token for the VSTS account in `~/git-credential-manager/insecureStore.xml`; it was imported in the GNOME Keyring and then the token was used to successfully fetch.
3. I removed a character from the token using "Seahorse" (the "Passwords and Keys" GUI to GNOME Keyring) to make it invalid and attempted another `git fetch`.
    1. The GCM4ML noticed the token was invalid and initiated authentication using Device Flow (the JVM is the stock openjdk version "1.8.0_91", so JavaFX was not available), which I completed in a web browser.  The fetch was again successful.
4. I also ran a quick smoke test on Mac OS X to confirm 

Mission accomplished!
